pkg/skaffold/build/custom/custom.go

```diff
@@ -89,20 +89,27 @@ func (b *ArtifactBuilder) retrieveCmd(out io.Writer, a *latest.Artifact, tag str
 	return cmd, nil
 }
 
-func (b *ArtifactBuilder) retrieveEnv(a *latest.Artifact, tag string) ([]string, error) {
-	buildContext, err := buildContext(a.Workspace)
+func (b *ArtifactBuilder) retrieveEnv(workspace string, a *latest.CustomArtifact, tag string) ([]string, error) {
```

Comment:
This name is at odds with `hash.go`'s `retrieveEnv` that literally just retrieves the appropriate env section.  This `retrieveEnv` doesn't seem quite right since there is generation happening here too.  `builderEnv()`?

Reply:
I've re-focused this PR on the buildpacks builder only